go/bls/bls.go

```diff
@@ -232,6 +243,57 @@ func (self *PublicKey) Destroy() {
 	C.destroy_public_key(self.ptr)
 }
 
+func toBuffer(data []byte) C.Buffer {
+	dataPtr, dataLen := sliceToPtr(data)
+	return C.Buffer{
+		ptr: dataPtr,
+		len: C.ulong(dataLen),
+	}
+}
+
+// Each epoch gets verified with a message,extraData,aggregate publickeys
+func BatchVerifyEpochs(messages []*Message, shouldUseCompositeHasher bool) error {
+	var verified C.bool
+	msg_len := len(messages)
+
+	// Allocate a contiguous slice of memory for our Messages
+	size := int(unsafe.Sizeof(C.MessageFFI{}))
+	list := C.malloc(C.size_t(size * msg_len))
+	for i := 0; i < msg_len; i++ {
+		// Get a reference to the memory where the i_th element will be at
+		ptr := unsafe.Pointer(uintptr(list) + uintptr(size*i))
+
+		data := toBuffer(messages[i].Data)
```

Comment:
Is this properly destroyed?

Reply:
As discussed, there's no allocation done here, so nothing to destroy